main.go

```diff
@@ -89,8 +90,17 @@ func main() {
 		log.SetOutput(f)
 	}
 
+	sigKillChan := make(chan os.Signal, 1)
```

Comment:
Should the process signal handling be done outside ProxyListener's receiver methods?

Here I've chosen to implement the signal handling inside Main whilest the ProxyListener library has a very simple API where the calling party calls StartListeners and then later calls StopListeners.